sunpy/map/sources/gong.py

```diff
@@ -0,0 +1,92 @@
+"""
+GONG Map subclass definitions
+"""
+import numpy as np
+
+import astropy.units as u
+from astropy.time import Time
+
+from sunpy.coordinates import HeliographicStonyhurst, get_earth
+from sunpy.map import GenericMap
+
+__all__ = ['GONGSynopticMap']
+
+
+class GONGSynopticMap(GenericMap):
+    """
+    GONG Synoptic Map.
+
+    The Global Oscillation Network Group (GONG) operates a six-station network of velocity
+    imagers located around the Earth that observe the Sun nearly continuously. GONG
+    produces hourly photospheric magnetograms using the Ni I 676.8 nm spectral line with an
+    array of 242×256 pixels covering the solar disk. These magnetograms are used to derive
+    synoptic maps which show a full-surface picture of the solar magnetic field.
+
+    Notes
+    -----
+    SunPy automatically fixes non-compliant FITS metadata in GONG maps. Namely, CDELT2 is
+    converted to degrees and CUNIT1 and CUNIT2 are set to 'deg'; DATE-OBS is converted to
+    ISO format; BUNIT is set to 'G'; and HGLT_OBS, HGLN_OBS, and DSUN_OBS are set to the
+    appropriate values for an observer on Earth.
+
+    References
+    ----------
+    * `GONG Page <https://gong.nso.edu/>`_
+    * `Magnetogram Synoptic Map Images Page <https://gong.nso.edu/data/magmap/>`_
+    * `FITS header keywords <https://gong.nso.edu/data/DMAC_documentation/General/fitsdesc.html>`_
+    * `Instrument Paper (pp. 203–208) <https://inis.iaea.org/collection/NCLCollectionStore/_Public/20/062/20062491.pdf>`_
+    * `GONG+ Documentation <https://gong.nso.edu/data/DMAC_documentation/PipelineMap/GlobalMap.html>`_
+
+
+    """
+    def __init__(self, data, header, **kwargs):
```

Comment:
I think the way this `__init__` is written isn't the way we want to handle this type of metadata fixing any more. These fixes should be implemented by overloading the properties on the `GenericMap` class rather than editing the `.header` property in the `__init__`.

For example you would "fix" CUNIT1/2 by writing a property like this: https://github.com/sunpy/sunpy/blob/main/sunpy/map/mapbase.py#L1298